### PR TITLE
gramine libfuzzer validation code

### PR DESCRIPTION
**Pull request by bprashan**

### CODE REVIEW

File: fuzzing/libfuzzer/common/config/system_packages.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
```
Comment (jinengandhi-intel) [inline fuzzing/libfuzzer/common/config/system_packages.yaml:1]: Prashanth you don't need so many packages to just build, install and run Gramine. follow this link: https://gramine.readthedocs.io/en/stable/devel/building.html#common-dependencies

and this libfuzzer we won't run on all distros, assume its Ubuntu22.04.

File: fuzzing/libfuzzer/conftest.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import pytest
+import os
+from libs import utils
+from libs import gramine_libs
+from common.config.constants import *
+
+@pytest.fixture(scope="session", autouse=True)
+def gramine_setup():
+    print_env_variables()
+    print("\n###### In gramerf_setup #####\n")
```
Comment (jinengandhi-intel) [inline fuzzing/libfuzzer/conftest.py:10]: Remove gramerf.

File: fuzzing/libfuzzer/conftest.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    # Delete old logs if any and create new logs directory.
+    if os.path.exists(LOGS_DIR):
+        del_logs_cmd = 'rm -rf ' + LOGS_DIR
+        os.system(del_logs_cmd)
+
+    os.makedirs(LOGS_DIR, exist_ok=True)
+
+    # Setting up the node environment and clearing cache.
+    utils.set_permissions()
+    utils.set_http_proxies()
```
Comment (jinengandhi-intel) [inline fuzzing/libfuzzer/conftest.py:21]: you don't need this. all machines will have this.

File: fuzzing/libfuzzer/libs/utils.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+    return config_dict
+
+
+def get_distro_and_version():
+    distro = lsb_release.get_distro_information().get('ID').lower()
+    distro_version = lsb_release.get_distro_information().get('RELEASE')
+    return distro, distro_version
+
+
+def clear_system_cache():
```
Comment (jinengandhi-intel) [inline fuzzing/libfuzzer/libs/utils.py:80]: this is for perf, for libfuzzer you don't need this.

File: fuzzing/libfuzzer/libs/gramine_fuzzing_libs.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    with open(os.path.join(LIBFUZZER_CORPUS_DIR, str(filesize)), 'wb') as file:
+        file.write(os.urandom(filesize))
+    print(utils.exec_shell_cmd('clang -g -fsanitize=fuzzer example.c -o example_fuzzer'))
+    print(utils.exec_shell_cmd('gramine-sgx-pf-crypt encrypt -w files/wrap_key -i corpus -o cipher_corpus'))
+    print(utils.exec_shell_cmd('make clean'))
+    print(utils.exec_shell_cmd('make SGX=1'))
+
+def run_libfuzzer(testname):
+    for i in range(5):
+        output = utils.exec_shell_popen('gramine-sgx libfuzz cipher_corpus', testname, 120)
```
Comment (jinengandhi-intel) [inline fuzzing/libfuzzer/libs/gramine_fuzzing_libs.py:17]: The timeout for each iteration, for ex: 120 in this case, have this as a Jenkins parameter. If nothing is passed then default is 120.

File: fuzzing/libfuzzer/libs/gramine_fuzzing_libs.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+def build_libfuzzer(filesize):
+    with open(os.path.join(LIBFUZZER_CORPUS_DIR, str(filesize)), 'wb') as file:
+        file.write(os.urandom(filesize))
+    print(utils.exec_shell_cmd('clang -g -fsanitize=fuzzer example.c -o example_fuzzer'))
+    print(utils.exec_shell_cmd('gramine-sgx-pf-crypt encrypt -w files/wrap_key -i corpus -o cipher_corpus'))
+    print(utils.exec_shell_cmd('make clean'))
+    print(utils.exec_shell_cmd('make SGX=1'))
+
+def run_libfuzzer(testname):
+    for i in range(5):
```
Comment (jinengandhi-intel) [inline fuzzing/libfuzzer/libs/gramine_fuzzing_libs.py:16]: Take the number of iterations as Jenkins parameter,  if nothing is passed then set the default to 5.

**Review by jinengandhi-intel — CHANGES_REQUESTED** [review]

File: fuzzing/libfuzzer/common/config/system_packages.yaml (hunk 1 above)
Comment (bprashan) [reply fuzzing/libfuzzer/common/config/system_packages.yaml:1]: Updated

File: fuzzing/libfuzzer/conftest.py (hunk 2 above)
Comment (bprashan) [reply fuzzing/libfuzzer/conftest.py]: removed

File: fuzzing/libfuzzer/conftest.py (hunk 3 above)
Comment (bprashan) [reply fuzzing/libfuzzer/conftest.py]: removed

File: fuzzing/libfuzzer/libs/utils.py (hunk 4 above)
Comment (bprashan) [reply fuzzing/libfuzzer/libs/utils.py]: removed

File: fuzzing/libfuzzer/libs/gramine_fuzzing_libs.py (hunk 5 above)
Comment (bprashan) [reply fuzzing/libfuzzer/libs/gramine_fuzzing_libs.py]: added environment variable

File: fuzzing/libfuzzer/libs/gramine_fuzzing_libs.py (hunk 6 above)
Comment (bprashan) [reply fuzzing/libfuzzer/libs/gramine_fuzzing_libs.py]: added environment variable

File: fuzzing/libfuzzer/common/config/constants.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+PYTHON_PACKAGES_FILE = "python_packages.yaml"
+PKG_INSTALL_WAIT_TIME = 25
+TEST_SLEEP_TIME_BW_ITERATIONS = 15
+BUILD_TYPE = "release"
+BUILD_PREFIX = FRAMEWORK_HOME_DIR + "/gramine_install/usr"
+
+# Commands constants
+GRAMINE_CLONE_CMD = "git clone https://github.com/gramineproject/gramine.git"
+GRAMINE_DEFAULT_REPO = "https://github.com/gramineproject/gramine.git"
+GRAMINE_CLONE = "RUN git clone --depth 1 --branch v1.6 https://github.com/gramineproject/gramine.git"
```
Comment (jinengandhi-intel) [inline fuzzing/libfuzzer/common/config/constants.py:24]: why are we hardcoding the clone command to always clone v1.6?

File: fuzzing/libfuzzer/common/config/python_packages.yaml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+# Need to enclose few of the below dependencies within quotes as the cmd string
+# will be formed with spaces and pip install will fail without the quotes.
+Default:
+ asv>=0.5.1
+ "meson >=0.56, <0.57"
+ "tomli>=1.1.0"
+ "tomli-w>=0.4.0"
+ openpyxl
+
+"18.04":
```
Comment (jinengandhi-intel) [inline fuzzing/libfuzzer/common/config/python_packages.yaml:11]: Like I had mentioned in my previous review, you don't need to support 18.04, you can remove this section.

File: fuzzing/libfuzzer/common/config/python_packages.yaml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+---
+# Need to enclose few of the below dependencies within quotes as the cmd string
+# will be formed with spaces and pip install will fail without the quotes.
+Default:
+ asv>=0.5.1
```
Comment (jinengandhi-intel) [inline fuzzing/libfuzzer/common/config/python_packages.yaml:5]: why do you need asv for libfuzzer?

File: fuzzing/libfuzzer/common/config/system_packages.yaml [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+ sshpass
+ sudo
+ texinfo
+ uthash-dev
+ vim
+ wget
+ zlib1g
+ zlib1g-dev
+ 
+"18.04":
```
Comment (jinengandhi-intel) [inline fuzzing/libfuzzer/common/config/system_packages.yaml:82]: Remove section for 18.04 and other distros.

**Review by jinengandhi-intel — CHANGES_REQUESTED** [review]

**Pull request closed without merging**